tests/helper.js

```diff
@@ -233,52 +221,39 @@ module.exports = {
    * selector : select element
    * item : option(s) in the dropdown.
    */
-  selectDropdownByText: (element, item, milliseconds) => {
-    element.all(by.tagName('option')).then(options => {
-      options.some(option => {
-        option.getText()
-          .then(text => {
-            if (text.indexOf(item) !== -1) {
-              option.click();
-            }
-          })
-          .catch(err => {
-            throw err;
-          });
-      });
-    }).catch(err => {
-      throw err;
-    });
+  electDropdownByText: async (element, item, milliseconds) => {
+    const options = await element.all(by.tagName('option'));
+    for (const option of options) {
+      const text = await option.getText();
+      if (text && text.includes(item)) {
+        await option.click();
+      }
+    }
     if (milliseconds) {
-      browser.sleep(milliseconds);
+      await browser.sleep(milliseconds);
     }
   },
 
-  selectDropdownByValue: (element, value, milliseconds) => {
-    element.all(by.css(`option[value="${value}"]`)).then(options => {
-      if (options[0]) {
-        options[0].click();
-      }
-    }).catch(err => {
-      throw err;
-    });
+  selectDropdownByValue: async (element, value, milliseconds) => {
+    const options = await element.all(by.css(`option[value="${value}"]`));
+    if (options[0]) {
+      await options[0].click();
+    }
     if (milliseconds) {
-      browser.sleep(milliseconds);
+      await browser.sleep(milliseconds);
```

Comment:
As a NIT and not change needed: Looks like we only ever pass 2 milliseconds into this. I don't understand what benefit that has. I would argue it would be better to `waitElementToDisappear(options[0]);`

Reply:
waitElementToDisappear(options[0]) always fails, apparently. Because this is a standard select, its options are always present.

Reply:
In the end, I went with a wait and checked which value is actually selected. The options themselves were always present/visible, for some reason.

Reply:
Yeah I can see that being a pain. Cool!